### DISCUSSION TITLE
Connect pgAdmin to my local supabase instance

### DISCUSSION THREAD

**Discussion by blizkyone** [supabase/supabase#6056] [Questions] [answered]
I'm Registering a Server in PG admin with the following info:
Host: localhost
Port: 54322
Username: 
Password: 

But I can't recall creating a username and password for my local instance

**Comment by GaryAustin1** [comment]
Info should be here:
![image](https://user-images.githubusercontent.com/54564956/160015883-a3f56be3-8606-4e26-b4a6-b98fb4e24d89.png)

You evidently can reset the password there, but I have not personally done that...

**Comment by imrim12** [comment, marked as answer]
@blizkyone you can access to your local supabase instance by setting these in PGAdmin when create a new server connection:
Host name/address: `locahost`
Port: `54322`
Database: `postgres`
Username: `postgres`
Password: `postgres`
These are broken by from the DB URL Supabase gave us:
![image](https://user-images.githubusercontent.com/46400321/170321100-43e89295-adfd-4001-bb5d-38c875d6ca1b.png)

**Reply by imrim12** [reply]
Make sure you also disable SSL in PGAdmin because by default we don't server through HTTPS
![image](https://user-images.githubusercontent.com/46400321/170321898-f13cbc35-f668-429f-9c33-8140878f9a80.png)

**Reply by hyunbinseo** [reply]
(Typo) The host name field should be `localhost`.

The following error message is shown, if there is a typo.

> connection is bad: nodename nor servname provided, or not known

**Reply by danya0365** [reply]
I can't make it work with following screenshot

DB URL: postgresql://postgres:postgres@localhost:54322/postgres

also disable ssl mode still can not reach it.

<img width="701" alt="Screen Shot 2566-07-14 at 15 40 06" src="https://github.com/supabase/supabase/assets/5410022/020fb205-37ae-4f9e-ab43-36eeaf97ac54">


<img width="706" alt="Screen Shot 2566-07-14 at 15 39 57" src="https://github.com/supabase/supabase/assets/5410022/d12a2a1a-f890-4ac4-ba3f-10d26286c0b9">

**Reply by alexmartinezm** [reply]
I ran into this issue as well. Basically, you have to open your docker and check which ports are open on your containers. Once you verify them, update your config.toml with the ports from docker and you're all set.

Ping me if you still struggle with this.

**Reply by longtimedeveloper** [reply]
@blizkyone thank you very much for your help. Your solution worked the first time, thank you.

**Comment by k-daw** [comment]
hello there 
i am trying to connect from pg-admin (on docker) to the supabse db ( on docker image)

I am getting the same error OR  a loading screen 

![image](https://github.com/supabase/supabase/assets/23188670/10249e38-bed0-4671-b1e5-1e84ea0445df)

"Basically, you have to open your docker and check which ports are open on your containers." 
On running 
``` lsof -i TCP:5432 ```
i get 
![image](https://github.com/supabase/supabase/assets/23188670/67d97317-5ee2-492d-ad4b-19d5d32892a8)

Any thoughts? 

i disabled ssl too

For reference (docker desktop)
![image](https://github.com/supabase/supabase/assets/23188670/3736517f-23e3-4d1f-8ad7-49479ff96786)

**Reply by k-daw** [reply]
Found the issue


you should set:

`host: host.docker.internal`

but i wonder why?
- at the end of the day, we have the port exposed 5432, why do we have to specify docker.host.internal?

but this is not the place for this question probably

**Reply by tzwtmll** [reply]
因为pgadmin的端口也是5432，将supabase/docker的端口占用了，所以这就是为什么supabase cli需要将5432切换到54322的原因

**Comment by consciousr** [comment]
And shouldn't the port be 5432?
Because there I see port 54322.
